main/src/main/java/org/mobilitydata/gtfsvalidator/validator/UrlConsistencyValidator.java

```diff
@@ -0,0 +1,239 @@
+                stop.stopUrl(),
+                agency.csvRowNumber()
+            ));
+      }
+    }
+    Map<String, GtfsRoute> routesByUrlMap = routesByUrlMap(routeTable);
+    Map<String, GtfsStop> stopsByUrlMap = stopsByUrlMap(stopTable);
+
+    Maps.filterEntries(stopsByUrlMap,
+        entry -> routesByUrlMap.get(entry.getValue().stopUrl()) != null)
+        .values().forEach(stopWithDuplicateUrl -> {
+      noticeContainer.addValidationNotice(
+          new SameStopAndRouteUrlNotice(
+              stopWithDuplicateUrl.csvRowNumber(),
+              stopWithDuplicateUrl.stopId(),
+              stopWithDuplicateUrl.stopUrl(),
+              routesByUrlMap.get(stopWithDuplicateUrl.stopUrl()).routeId(),
+              routesByUrlMap.get(stopWithDuplicateUrl.stopUrl()).csvRowNumber()
+          ));
+    });
+  }
+
+  /**
+   * Maps {@code GtfsAgency}s by there URLs if provided.
+   *
+   * @param stopTable the {@code GtfsStopTableContainer} to extract {@code GtfsStop} from
+   * @return routes from {@code GtfsStopTableContainer}s mapped by there {@code routes.route_url}
+   * (in lower case) if provided.
+   */
+  private Map<String, GtfsStop> stopsByUrlMap(GtfsStopTableContainer stopTable) {
+    Map<String, GtfsStop> stopsByUrl = new HashMap<>();
+    stopTable.getEntities().forEach(stop -> {
+      if (stop.hasStopUrl()) {
+        if (stopsByUrl.get(stop.stopUrl()) == null) {
+          stopsByUrl.put(stop.stopUrl().toLowerCase(), stop);
+        }
+      }
+    });
+    return stopsByUrl;
+  }
+
+  /**
+   * Maps {@code GtfsRoute}s by their URLs if provided.
+   *
+   * @param routeTable the {@code GtfsRouteTableContainer} to extract {@code GtfsRoute} from
+   * @return routes from {@code GtfsRouteTableContainer}s mapped by there {@code routes.route_url}
+   * (in lower case) if provided.
+   */
+  private Map<String, GtfsRoute> routesByUrlMap(GtfsRouteTableContainer routeTable) {
+    Map<String, GtfsRoute> routesByUrl = new HashMap<>();
+    routeTable.getEntities().forEach(route -> {
+      if (route.hasRouteUrl()) {
+        if (routesByUrl.get(route.routeUrl()) == null) {
+          routesByUrl.put(route.routeUrl().toLowerCase(), route);
+        }
+      }
+    });
+    return routesByUrl;
+  }
+
+  /**
+   * Maps {@code GtfsAgency}s by their URLs if provided.
+   *
+   * @param agencyTable the {@code GtfsAgencyTableContainer} to extract {@code GtfsAgency} from
+   * @return agencies from {@code GtfsAgencyTableContainer}s mapped by there URLs (in lower case) if
+   * provided.
+   */
+  private Map<String, GtfsAgency> agenciesByUrlMap(GtfsAgencyTableContainer agencyTable) {
+    Map<String, GtfsAgency> agenciesByUrl = new HashMap<>();
+    agencyTable.getEntities().forEach(agency -> {
+      if (agency.hasAgencyUrl()) {
+        if (agenciesByUrl.get(agency.agencyUrl()) == null) {
+          agenciesByUrl.put(agency.agencyUrl().toLowerCase(), agency);
+        }
+      }
+    });
+    return agenciesByUrl;
+  }
```

Comment:
As discussed, it might not be the best way to proceed but since the `GtfsEntity` do not have a unified interface that provides an implementation for `url()` method it would be difficult to apply polymorphism here. Hence we'll continue with the current implementation.